### PR TITLE
Refactor cases to analyze

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

- Module to reuse SQL filtering operations 
- Add test for module above

### Changed

- Split query to separate function
- Refactor cases_to_analyze to use filtering functions instead of hard coded filtering operations
- Refactor tests
- Moved utility function to helper class for future reuse

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh refactor_cases_to_analyze`

### How to test
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by KN
- [x] tests executed by HS
- [x] "Merge and deploy" approved by HS
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
### cg status analysis ✅ 
```
cg status analysis
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
wiseiguana (F0013549)
wantedrodent (F0006360)
helpedsatyr (WNB124)
betterbison (GSU492)
freeelf (GSU493)
quickmartin (417701600682fam)
primequail (817701600683fam)
worthybedbug (F0031832)
actualjoey (EU003597N)
moremonkey (F0033384)
rapidsatyr (F0015829)
comicpanda (F0033411)
helpedcougar (F0033543)
restedamoeba (F0033545)
cuddlyowl (F0033642)
amusedhusky (F0033477)
unitedstag (F0031362)
prettyhusky (F0000128-2)
soughtrhino (20078)
promptjaguar (F0033174-2)
caringbuck (F0034774)
noblerodent (F0019576)
clearcow (F0028201)
boldbengal (720701400155fam)
funnycamel (F0035698)
toughkit (F0032870)
mainhog (F0035919)
humaneosprey (F0035370-3)
comicpuma (F0012168)
loyalsole (F0022528)
upwardcow (F0033300-1)
coolcrane (F0011847-2)
cutetrout (F0035446)
lighttomcat (F0036128)
moralmoth (F0036200)
primechow (F0030552)
nativewalrus (G000309532-20200923)
sweetmartin (F0032413)
warmcicada (F0035043-3)
hardyfeline (F0036477)
soughtimpala (F0010594-2)
viablewhale (F0036516)
viableegret (F0005406-2)
hardygopher (F0028433-2)
inpig (F0036759)
loyaltoucan (F0035509)
mainpoodle (F0036645)
promptswine (F0036081)
usefuleft (F0036742)
hardyspider (F0011711)
manygibbon (F0017292)
trustysloth (F0031637)
grownewe (F0036765)
solidpika (F0034874-2)
smartraven (F0035463-2)
prettymole (F0030374)
usablelion (20215)
decentpika (20214)
livingalien (F0027356-3)
robustcoral (F0033756)
kindwhale (F0011690)
strongcaiman (F0035846)
busyshrimp (F0037644)
tightferret (F0037667)
modelrabbit (F0037412-2)
helpedteal (F0037825)
caringjaguar (F0037826)
notedmarlin (F0037870)
crispspider (F0019055)
tightyeti (F0031464)
alertseal (F0019229)
sharpdeer (F0037934)
fairferret (F0034141)
savinglouse (F0033165)
opencollie (21002)
worthyzebra (F0038035)
livingjay (21028)
squarecattle (F0038489)
happyelf (F0033615-1)
activetarpon (F0036316-1)
fitbird (F0038479)
cleverkodiak (F0038481)
handycow (F0035587)
superbunny (F0016706)
finercivet (F0031670-1)
goldenskink (F0038645-1)
unitedbug (F0038898)
maincod (F7)
robusttrout (F16)
fluentman (F17)
grandpiglet (F22)
proboar (F41)
safejoey (F54)
livingdragon (F61)
nativepuma (F0039626)
holyamoeba (F0039587)
raretomcat (F0033533)
guidedrhino (F0039640)
promptfilly (F0034511-2)
majormullet (F0033829)
jointmutt (F0037368)
masterdane (F0039950)
loyalstud (F0039220)
richchow (F0022432-3)
deepjaguar (F0039446-3)
clearrobin (F0036518)
majormoose (F0040334)
upbaboon (F0030903)
poeticdane (F0038399-1)
usefulsponge (F0035128-1)
goldentick (F0039524-1)
pumpedshiner (F0036897-1)
valuedweevil (F0038210-1)
usableraptor (F0039144-1)
fitkoi (F0038446-1)
simplealpaca (F0039460-1)
pumpedbass (F0040002-1)
validsalmon (F0039101-1)
warmdassie (F0040669)
richmartin (F0040691)
livehusky (F0040706)
usefulmarlin (F0039792)
readyrhino (F0040147)
securestag (F0040840)
merryjay (18005-2)
kindbird (F0035127)
mutualpanda (F0041172)
warmakita (F0040265)
uniquesawfly (F0041266)
novelrodent (F0030071-2)
morelion (F0034694-2)
immuneworm (F0042085)
knownhusky (21341-trio)
gentlekrill (221701400279fam)
luckybuck (21354)
dearkit (21213-4)
fitminnow (21242-trio)
immunemutt (21356)
wittyshrimp (21357)
superbgibbon (21358)
upwardmarlin (21359)
steadytoucan (21360)
suitedshad (21361)
subtlerat (21362)
pumpedrodent (21363)
savedbee (21364)
fonddingo (21365)
finerteal (21315-trio)
comicgoose (21366)
civilvervet (F0039781-2)
fitcollie (F0043660)
cutepoodle (F0040256)
deeptarpon (F0043633)
suitedspider (F0043632)
realiguana (F0043663)
hipasp (F0043641)
vitaldoe (F0043654)
morebee (F0043661)
suitedlion (F0043655)
crispwhale (F0043670)
steadyoriole (F0042110)
welldrake (F0043695)
fullcrab (F0043694)
truedragon (F0043722)
expertguinea (F0001505-5)
movingkodiak (GMSB-UM010MIP)
wittyfeline (F0042957)
vocalchow (F0039964)
restedtoucan (F0042980)
comicdove (F0045099)
onsatyr (F0045120)
realimp (F0040237)
epicsawfly (F0045110)
majorpiglet (F0045402)
savingturkey (F0042834-2)
easyaphid (F0040177)
sunnyboxer (F0018767)
bosstuna (F0008420)
realbass (F0007832-2)
uniquedodo (F0014337-2)
rulingfox (F0014287)
provencalf (F0047367)
daringpoodle (F0041421)
heroicstork (822701400209fam)
trustyroughy (22257)
boldjoey (F0043087-2)
lovingtomcat (22204-trio)
puremuskox (21297-trio)
topbison (F0039607-2)
keenjackal (F0009865-3)
modelsawfly (F0042526-2)
```

No logic should have changed so I think this is sufficient as a test. Note that this CLI command only will show pipeline=Pipeline.MIP_DNA as the pipeline is hard coded in the CLI call (should be changed in the future)

**Review by karlnyr — APPROVED** [review]
This is great! Way easier to understand the query now, and more test coverage of our statushander is well needed! 🔥

File: cg/store/api/status.py [hunk 1]
```diff
@@ -66,52 +67,43 @@ def samples_to_sequence(self) -> Query:
             .order_by(models.Sample.received_at)
         )
 
+    def get_families_with_extended_models(self) -> Query:
+        """Return all cases in the database with an analysis"""
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/store/api/status.py]: Fixed

**Comment by henrikstranneheim** [comment]
@karlnyr Thx for the review! I will go ahead and M&D this now

**Comment by henrikstranneheim** [comment]
Deployed on Hasta:
```
[henrik.stranneheim@hasta:~] [P_main] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-prod.sh
on the host hasta.scilifelab.se, continue
This will install the latest version of clinical-genomics/cg on master in production, are you sure? y/[n]y
Collecting git+https://github.com/clinical-genomics/cg
  Cloning https://github.com/clinical-genomics/cg to ./tmp/pip-req-build-c1g1jcsk
  Running command git clone -q https://github.com/clinical-genomics/cg /home/henrik.stranneheim/tmp/pip-req-build-c1g1jcsk
  Resolved https://github.com/clinical-genomics/cg to commit 45c915c20c30a5cd1dd57f490c6769a8a31b9a01
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: Click<7.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (6.7)
Requirement already satisfied: Flask>=1.1.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.1.1)
Requirement already satisfied: housekeeper>=2.5 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (3.2.0)
Requirement already satisfied: alive-progress in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.6.2)
Requirement already satisfied: Flask-CORS in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (3.0.7)
Requirement already satisfied: requests-cache in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.4.13)
Requirement already satisfied: petname in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.2)
Requirement already satisfied: Flask-Alchy in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.5.0)
Requirement already satisfied: python-dateutil in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.8.0)
Requirement already satisfied: typing-extensions in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (3.10.0.2)
Requirement already satisfied: cgmodels>=0.10.2 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.10.3)
Requirement already satisfied: Flask-Dance in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.3.0)
Requirement already satisfied: Alchy in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.2.2)
Requirement already satisfied: markupsafe<2.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.1.0)
Requirement already satisfied: genologics in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.4.3)
Requirement already satisfied: requests[security] in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.21.0)
Requirement already satisfied: pyyaml in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (5.1)
Requirement already satisfied: setuptools>=39.2.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (40.6.2)
Requirement already satisfied: werkzeug<1.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.16.1)
Requirement already satisfied: gunicorn in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (20.0.4)
Requirement already satisfied: ansi in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.2.0)
Requirement already satisfied: openpyxl in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.6.0)
Requirement already satisfied: packaging in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (20.9)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.1)
Requirement already satisfied: tabulate in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.8.3)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.2.17)
Requirement already satisfied: sendmail-container in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.2.0)
Requirement already satisfied: wtforms<3.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.0.5)
Requirement already satisfied: google-auth in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.6.2)
Requirement already satisfied: Flask-Admin in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.5.3)
Requirement already satisfied: lxml in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (4.3.1)
Requirement already satisfied: pymysql in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (0.9.3)
Requirement already satisfied: coloredlogs in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (10.0)
Requirement already satisfied: pydantic in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.9.1)
Requirement already satisfied: paramiko in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (2.7.2)
Requirement already satisfied: pandas in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.3.4)
Requirement already satisfied: marshmallow>3 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (3.6.1)
Requirement already satisfied: blinker in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.4)
Requirement already satisfied: email-validator in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cg==22.77.2) (1.1.2)
Requirement already satisfied: pluggy in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (1.0.0)
Requirement already satisfied: pytest in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (7.1.2)
Requirement already satisfied: more-itertools in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (6.0.0)
Requirement already satisfied: dataclasses in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (0.6)
Requirement already satisfied: pytest-cov in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (3.0.0)
Requirement already satisfied: attrs in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (21.4.0)
Requirement already satisfied: atomicwrites in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (1.3.0)
Requirement already satisfied: colorama in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (0.4.3)
Requirement already satisfied: coverage in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (6.3.2)
Requirement already satisfied: importlib-metadata in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (4.11.3)
Requirement already satisfied: py in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (1.11.0)
Requirement already satisfied: pyparsing in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (2.4.7)
Requirement already satisfied: zipp in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (3.8.0)
Requirement already satisfied: wcwidth in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cgmodels>=0.10.2->cg==22.77.2) (0.1.7)
Requirement already satisfied: itsdangerous>=0.24 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask>=1.1.1->cg==22.77.2) (1.1.0)
Requirement already satisfied: Jinja2>=2.10.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask>=1.1.1->cg==22.77.2) (2.11.1)
Requirement already satisfied: ruamel.yaml in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from housekeeper>=2.5->cg==22.77.2) (0.15.88)
Requirement already satisfied: rich in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from housekeeper>=2.5->cg==22.77.2) (6.2.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Alchy->cg==22.77.2) (4.7.4)
Requirement already satisfied: humanfriendly>=4.7 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from coloredlogs->cg==22.77.2) (4.17)
Requirement already satisfied: idna>=2.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from email-validator->cg==22.77.2) (2.8)
Requirement already satisfied: dnspython>=1.15.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from email-validator->cg==22.77.2) (2.1.0)
Requirement already satisfied: Six in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask-CORS->cg==22.77.2) (1.13.0)
Requirement already satisfied: oauthlib in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==22.77.2) (2.1.0)
Requirement already satisfied: lazy in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==22.77.2) (1.4)
Requirement already satisfied: urlobject in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==22.77.2) (2.4.3)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from Flask-Dance->cg==22.77.2) (1.1.0)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==22.77.2) (0.2.4)
Requirement already satisfied: cachetools>=2.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==22.77.2) (3.1.0)
Requirement already satisfied: rsa>=3.1.4 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from google-auth->cg==22.77.2) (4.0)
Requirement already satisfied: jdcal in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from openpyxl->cg==22.77.2) (1.4)
Requirement already satisfied: et-xmlfile in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from openpyxl->cg==22.77.2) (1.0.1)
Requirement already satisfied: numpy>=1.17.3 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pandas->cg==22.77.2) (1.21.4)
Requirement already satisfied: pytz>=2017.3 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pandas->cg==22.77.2) (2018.9)
Requirement already satisfied: bcrypt>=3.1.3 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==22.77.2) (3.2.0)
Requirement already satisfied: pynacl>=1.0.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==22.77.2) (1.4.0)
Requirement already satisfied: cryptography>=2.5 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from paramiko->cg==22.77.2) (2.5)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==22.77.2) (2018.10.15)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==22.77.2) (3.0.4)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==22.77.2) (1.24.1)
Requirement already satisfied: pyOpenSSL>=0.14 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from requests[security]->cg==22.77.2) (19.0.0)
Requirement already satisfied: uvicorn in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==22.77.2) (0.5.1)
Requirement already satisfied: fastapi in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==22.77.2) (0.78.0)
Requirement already satisfied: python-multipart in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from sendmail-container->cg==22.77.2) (0.0.5)
Requirement already satisfied: cffi>=1.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from bcrypt>=3.1.3->paramiko->cg==22.77.2) (1.12.0)
Requirement already satisfied: asn1crypto>=0.21.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cryptography>=2.5->paramiko->cg==22.77.2) (0.24.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth->cg==22.77.2) (0.4.5)
Requirement already satisfied: starlette==0.19.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from fastapi->sendmail-container->cg==22.77.2) (0.19.1)
Requirement already satisfied: anyio<5,>=3.4.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from starlette==0.19.1->fastapi->sendmail-container->cg==22.77.2) (3.6.1)
Requirement already satisfied: python-dotenv>=0.10.4 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pydantic->cg==22.77.2) (0.20.0)
Requirement already satisfied: iniconfig in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pytest->cgmodels>=0.10.2->cg==22.77.2) (1.1.1)
Requirement already satisfied: tomli>=1.0.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from pytest->cgmodels>=0.10.2->cg==22.77.2) (2.0.1)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from rich->housekeeper>=2.5->cg==22.77.2) (0.9.1)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from rich->housekeeper>=2.5->cg==22.77.2) (2.6.1)
Requirement already satisfied: websockets>=6.0 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==22.77.2) (10.3)
Requirement already satisfied: h11 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==22.77.2) (0.9.0)
Requirement already satisfied: httptools in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==22.77.2) (0.4.0)
Requirement already satisfied: uvloop in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from uvicorn->sendmail-container->cg==22.77.2) (0.16.0)
Requirement already satisfied: pycparser in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from cffi>=1.1->bcrypt>=3.1.3->paramiko->cg==22.77.2) (2.19)
Requirement already satisfied: sniffio>=1.1 in /home/proj/bin/conda/envs/P_main/lib/python3.7/site-packages (from anyio<5,>=3.4.0->starlette==0.19.1->fastapi->sendmail-container->cg==22.77.2) (1.2.0)
Building wheels for collected packages: cg
  Building wheel for cg (pyproject.toml) ... done
  Created wheel for cg: filename=cg-22.77.2-py3-none-any.whl size=618216 sha256=152d0c8acc0cba83a01c402909f1a7b72a6fdd420389afe387283fa5b0d8ae40
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-dufspadm/wheels/8c/1e/b8/816ed4facf8c5628b55966d3f2448eeb6ca70f35deff59389e
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 22.77.1
    Uninstalling cg-22.77.1:
      Successfully uninstalled cg-22.77.1
Successfully installed cg-22.77.2
WARNING: You are using pip version 21.3.1; however, version 22.2.2 is available.
You should consider upgrading via the '/home/proj/bin/conda/envs/P_main/bin/python -m pip install --upgrade pip' command.
remote: Enumerating objects: 111, done.
remote: Counting objects: 100% (104/104), done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 54 (delta 41), reused 39 (delta 30), pack-reused 0
Unpacking objects: 100% (54/54), done.
From https://github.com/Clinical-Genomics/cg
   3f01df1..45c915c  master     -> origin/master
   786a4c3..c30148d  json_api   -> origin/json_api
 * [new tag]         v22.77.2   -> v22.77.2
Updating 3f01df1..45c915c
Fast-forward
 .bumpversion.cfg                                  |   2 +-
 cg/__init__.py                                    |   2 +-
 cg/constants/constants.py                         |   5 +++
 cg/constants/priority.py                          |   7 +++
 cg/store/api/status.py                            |  60 +++++++++++--------------
 cg/store/status_case_filters.py                   |  47 +++++++++++++++++++
 cg/store/utils.py                                 |   0
 setup.py                                          |   2 +-
 tests/store/api/conftest.py                       |  26 +++++++++--
 tests/store/api/test_store_api_status_analysis.py | 325 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------
 tests/store/test_status_cases_filters.py          | 276 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 tests/store_helpers.py                            |  65 ++++++++++++++++++++++-----
 12 files changed, 647 insertions(+), 170 deletions(-)
 create mode 100644 cg/store/status_case_filters.py
 delete mode 100644 cg/store/utils.py
 create mode 100644 tests/store/test_status_cases_filters.py
/home/henrik.stranneheim
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/henrik.stranneheim
done.
Log deploy... done.
Running --version on each Clinical-Genomics developed tool to verify that all requirements are met:
balsamic 6.0.0 requires colorclass, which is not installed.
argcomplete 1.12.1 has requirement importlib-metadata<3,>=0.23, but you have importlib-metadata 4.11.3.
cg, version 22.77.2
housekeeper, version 3.2.0
demux, version 5.10.4
cglims, version 1.3.4
```

**Pull request merged**